src/dev/encoder.cpp

```diff
@@ -0,0 +1,152 @@
+#include <EVT/dev/encoder.hpp>
+
+namespace EVT::core::DEV {
+
+Encoder::Encoder(IO::GPIO& a, IO::GPIO& b, int64_t range, int64_t initialPosition) : a(a), b(b), range(range), position(initialPosition) {
+    //making sure range is positive
+    if (range < 0) {
+        range *= -1;
+    }
+    setPosition(position);
+    //setting other instance variables
+    currentDirection = Static;
+    noChangeCounter = 0;
+    currentRelPos = readPinValues();
+}
+
+int64_t Encoder::getPosition() {
+    return position;
+}
+
+int8_t Encoder::update() {
+    //Reads current pin values
```

Comment:
Inline comments should generally be written as commands
```suggestion
    //Read current pin values
```